examples/stub.php

```diff
@@ -376,17 +375,15 @@ public function shutdown() {}
      * @link http://www.php.net/manual/en/worker.stack.php
      * @return int The new length of the stack
      */
-    public function stack(Threaded &$work) {}
+    public function stack(Threaded $work) {}
 
     /**
-     * Removes the referenced object ( or all objects if parameter is null ) from stack of the referenced Worker
-     *
-     * @param Threaded $work Threaded object previously stacked onto Worker
+     * Removes the first task (the oldest one) in the stack.
      *
      * @link http://www.php.net/manual/en/worker.unstack.php
-     * @return int The new length of the stack
+     * @return Collectable|null The item removed from the stack
```

Comment:
Perhaps it would be best to use `Threaded` here, rather than `Collectable`, since we use `Threaded` for `Worker::stack`, as well as `Pool::submit` and `Pool::submitTo`.

Reply:
On a semi-related note, the existence of `Collectable` as an interface is confusing as it isn't consistently used... it's also only implemented by `Threaded` and does not need to be implemented in userland (Volatile descends from Threaded anyway). To be honest it seems superfluous.

Reply:
Agreed. Perhaps that's something that can be removed in the next major version.